instrumentation/mongo/mongo-async-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/mongoasync/v3_3/MongoAsyncClientTest.java

```diff
@@ -0,0 +1,250 @@
+            MongoClientSettings.builder()
+                .clusterSettings(
+                    ClusterSettings.builder()
+                        .description("some-description")
+                        .applyConnectionString(
+                            new ConnectionString("mongodb://" + host + ":" + port))
+                        .build())
+                .build());
+  }
+
+  @AfterAll
+  void cleanup() {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+  @Override
+  protected InstrumentationExtension testing() {
+    return testing;
+  }
+
+  @Override
+  public void createCollection(String dbName, String collectionName) {
+    MongoDatabase db = client.getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public void createCollectionNoDescription(String dbName, String collectionName) {
+    MongoDatabase db = MongoClients.create("mongodb://" + host + ":" + port).getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public void createCollectionWithAlreadyBuiltClientOptions(String dbName, String collectionName) {
+    MongoClientSettings clientSettings = client.getSettings();
+    MongoClientSettings newClientSettings = MongoClientSettings.builder(clientSettings).build();
+    MongoDatabase db = MongoClients.create(newClientSettings).getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public void createCollectionCallingBuildTwice(String dbName, String collectionName) {
+    MongoClientSettings.Builder settings =
+        MongoClientSettings.builder()
+            .clusterSettings(
+                ClusterSettings.builder()
+                    .description("some-description")
+                    .applyConnectionString(new ConnectionString("mongodb://" + host + ":" + port))
+                    .build());
+    settings.build();
+    MongoDatabase db = MongoClients.create(settings.build()).getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public int getCollection(String dbName, String collectionName) {
+    MongoDatabase db = client.getDatabase(dbName);
+    CompletableFuture<Long> count = new CompletableFuture<>();
+    db.getCollection(collectionName).count(toCallback(o -> count.complete(((Long) o))));
+    return count.join().intValue();
+  }
+
+  @Override
+  public MongoCollection<Document> setupInsert(String dbName, String collectionName)
+      throws InterruptedException {
+    MongoCollection<Document> collection =
+        testing()
+            .runWithSpan(
+                "setup",
+                () -> {
+                  MongoDatabase db = client.getDatabase(dbName);
+                  CountDownLatch latch = new CountDownLatch(1);
+                  db.createCollection(collectionName, toCallback(result -> latch.countDown()));
+                  latch.await();
```

Comment:
Elsewhere `await` is used with timeout, perhaps we should add a timeout to await usages in this class too. In case of a failure the timeout makes the test fail in a much nicer way.